src/Mod/RuleSoldier.h

```diff
@@ -21,15 +21,38 @@
 #include <yaml-cpp/yaml.h>
 #include "Unit.h"
 #include "../Engine/Script.h"
+#include "../Mod/RuleItem.h"
+#include "../Battlescape/BattlescapeGame.h"
 
 namespace OpenXcom
 {
 
+enum BattleActionType : Uint8;
 class Mod;
 class ModScript;
 class SoldierNamePool;
 class StatString;
-class RuleItem;
+
+struct RuleSkill
+{
+	std::string Name;
+	RuleItemUseCost Cost;
+	RuleItemUseCost Flat;
+	BattleActionType TargetMode;
+	std::vector<std::string> CompatibleWeapons;
+	std::vector<std::string> RequiredBonus;
+	bool IsPsiRequired;
+	
+	bool CheckHandsOnly;
+	const int Id;
```

Comment:
I do not think this is correct way to recognize what skill is used, image you want have similar skill in multiple unit types.

Reply:
Skill definitions are rather short. They either handle an item action and would just be copy pastable or they would be completely scripted. In the latter case the script would what unit type activated which skill and could just call the right functions.

But I see, this goes hand in hand with your proposal of having the skills be a separate rule object which would indeed enable some more flexible use cases. So I agree.